# Generated by Django 3.0.5 on 2020-05-01 18:57

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ('threads', '0006_auto_20200501_2057'),
        ('users', '0001_initial'),
    ]

    operations = [
        migrations.CreateModel(
            name='ReplyCategoryModel',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=50)),
                ('owner', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
                ('replies', models.ManyToManyField(blank=True, related_name='linked_categories', to='threads.ReplyModel')),
            ],
        ),
        migrations.CreateModel(
            name='FactionModel',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=50)),
                ('leader', models.ForeignKey(null=True, on_delete=django.db.models.deletion.SET_NULL, to=settings.AUTH_USER_MODEL)),
                ('mutualy_exclusive_with', models.ManyToManyField(blank=True, related_name='_factionmodel_mutualy_exclusive_with_+', to='users.FactionModel')),
            ],
        ),
        migrations.AddField(
            model_name='usermodel',
            name='factions',
            field=models.ManyToManyField(related_name='members', to='users.FactionModel'),
        ),
    ]
